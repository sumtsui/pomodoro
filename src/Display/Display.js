import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ section, remaining, isWorking }) => {
  // console.log('Display Rendering!');
  return (
    <div className='display'>
      <h1>
        {
          isWorking ?
          `Section ${section}`
          :
          'Taking a break...'
        }
      </h1>
      <span className='count-down'>{remaining}</span>
    </div>
  )
}

Display.prototype = {
  section: PropTypes.string.isRequired,
  isWorking: PropTypes.bool.isRequired,
  remaining: PropTypes.string.isRequired
}

export default Display;