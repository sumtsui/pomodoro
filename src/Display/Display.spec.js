import React from 'react';
import Display from './Display';
import {shallow} from 'enzyme';

describe('Display', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Display />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1)
  })

  it('should display "Taking a break..." if isWorking is false', () => {
    wrapper.setProps({isWorking: false})
    expect(wrapper.find('.display h1').text()).toEqual('Taking a break...');
  })

  it('should show Section Number and Remaining Time if isWorking is true', () => {
    const section = 5;
    const remaining = '00:30:00';
    wrapper.setProps({
      section,
      remaining,
      isWorking: true
    });
    expect(wrapper.find('.display').text()).toEqual('Section ' + section + remaining);
  })
})