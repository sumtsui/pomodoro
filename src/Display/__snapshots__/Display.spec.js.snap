// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Taking a break...
</h1>,
        <span
          className="count-down"
/>,
      ],
      "className": "display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Taking a break...",
        },
        "ref": null,
        "rendered": "Taking a break...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "count-down",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Taking a break...
</h1>,
          <span
            className="count-down"
/>,
        ],
        "className": "display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Taking a break...",
          },
          "ref": null,
          "rendered": "Taking a break...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "count-down",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
